src/python-bindings/classad2/exprtree.cpp

```diff
@@ -141,24 +141,26 @@ _exprtree_simplify( PyObject *, PyObject * args ) {
     // we thus don't have to free/delete `v`.
     //
     // Not sure why this has to be so hard, but let's not mess with it.
-    classad::Value * v = NULL;
-    classad::Literal * literal = classad::Literal::MakeUndefined(v);
-    if(! evaluate( e, s, t, *v )) {
+    classad::Value v;
+    v.SetUndefinedValue();
+    if(! evaluate( e, s, t, v )) {
         // In version 1, this was ClassAdEvaluationError.
         PyErr_SetString(PyExc_RuntimeError, "failed to evaluate expression");
         return NULL;
     }
+	classad::Literal * literal = classad::Literal::MakeLiteral(v);
+
```

Comment:
The creation and deletion of literal should be inside  the default case. It's not valid to create a literal from a Value that's an ad or a list. This code will work correctly, since MakeLiteral() will return nullptr for an ad or list, but it's bad form.

Reply:
You forgot to move the delete of literal into the default case.